lib/Parse/ParseDecl.cpp

```diff
@@ -2406,10 +2406,24 @@ Decl *Parser::ParseDeclarationAfterDeclaratorAndAttributes(
           StopTokens.push_back(tok::r_paren);
         SkipUntil(StopTokens, StopAtSemi | StopBeforeMatch);
         Actions.ActOnInitializerError(ThisDecl);
-      } else
+      } else {
+        if (D.getDeclSpec().isUnpackSpecified()) {
+          // TODO: make more robust, add asserts, etc.
+          auto TpeVars = D.getDeclSpec().typeVariables();
+          assert(TpeVars.size() == 1 && "Expected exactly one type variable in _Unpack specifier");
+          auto TVar = Actions.getASTContext().getTypedefType(TpeVars[0], QualType());
+          auto *SourceInfo = Actions.Context.getTrivialTypeSourceInfo(TVar, TpeVars[0]->getLocation());
+          auto TypeArg = TypeArgument { TVar, SourceInfo };
+          auto *InitExpr = Init.get();
+          // TODO: typing the lhs might fail: need to handle gracefully.
+          auto InitType = cast<ExistentialType>(InitExpr->getType().getTypePtr())->innerType();
```

Comment:
This needs to be cleaned up too, as it could a compiler crash.

Reply:
Done